@here/harp-vectortile-datasource/lib/VectorTileDataEmitter.ts

```diff
@@ -1388,288 +1414,283 @@ export class VectorTileDataEmitter {
+                // Calculate nextEdge and nextWall.
+                vertices.push(
+                    properEdge ? nextIdx : -1,
+                    boundaryWalls || properEdge ? nextIdx : -1
+                );
+            }
 
-                // Iterate over the inner rings. The inner rings have the opposite winding
-                // of the outer rings.
-                const holes: number[] = [];
-                while (ringIndex < polygon.length && polygon[ringIndex].winding !== winding) {
-                    const vertexOffset = vertices.length / vertexStride;
-                    holes.push(vertexOffset);
+            // Iterate over the inner rings - if any
+            const holes: number[] = [];
+            let ringIndex = 1;
+            while (ringIndex < polygon.length) {
+                const vertexOffset = vertices.length / vertexStride;
+                holes.push(vertexOffset);
 
-                    const hole = polygon[ringIndex++];
-                    for (let i = 0; i < hole.points.length; ++i) {
-                        const nextIdx = (i + 1) % hole.points.length;
-                        const point = hole.points[i];
+                const hole = polygon[ringIndex++];
+                for (let i = 0; i < hole.points.length; ++i) {
+                    const nextIdx = (i + 1) % hole.points.length;
+                    const point = hole.points[i];
 
-                        // Invert the Y component to preserve the correct winding without
-                        // transforming from webMercator's local to global space.
-                        vertices.push(point.x, -point.y);
+                    // Invert the Y component to preserve the correct winding without
+                    // transforming from webMercator's local to global space.
+                    vertices.push(point.x, -point.y);
 
-                        if (hole.textureCoords !== undefined) {
-                            vertices.push(hole.textureCoords[i].x, hole.textureCoords[i].y);
-                        }
+                    if (hole.textureCoords !== undefined) {
+                        vertices.push(hole.textureCoords[i].x, hole.textureCoords[i].y);
+                    }
 
-                        // Calculate nextEdge and nextWall.
-                        const insideExtents = hole.isProperEdge(i);
+                    // Calculate nextEdge and nextWall.
+                    const insideExtents = hole.isProperEdge(i);
 
-                        vertices.push(
-                            insideExtents ? vertexOffset + nextIdx : -1,
-                            boundaryWalls || insideExtents ? vertexOffset + nextIdx : -1
-                        );
-                    }
+                    vertices.push(
+                        insideExtents ? vertexOffset + nextIdx : -1,
+                        boundaryWalls || insideExtents ? vertexOffset + nextIdx : -1
+                    );
                 }
+            }
 
```

Comment:
wow,  the diff got totally messed up beyond this point. I assume there's no changes beyond this point (apart from indentation) is that right?

Reply:
correct, nothing to check beyond here